server/src/handler_driver.c

```diff
@@ -168,6 +192,412 @@ static bool drive_route_params_valid(const char *train_id, t_interlocking_route
 	return true;
 }
 
+static bool validate_interlocking_route_members_not_null(const t_interlocking_route *route) {
+	if (route == NULL) {
+		return false;
+	}
+	return (route->conflicts != NULL && route->destination != NULL && route->id != NULL 
+	        && route->orientation != NULL && route->path != NULL && route->points != NULL 
+	        && route->sections != NULL && route->signals != NULL && route->source != NULL
+			  && route->train != NULL);
+}
+
+
+static void print_signal_info_member(int priority, const t_route_signal_info *sig_info) {
+	if (sig_info != NULL) {
+		syslog_server(priority, "\tID %s", (sig_info->id != NULL ? sig_info->id : "NULL") );
+		syslog_server(priority, "\tis source?      %s", sig_info->is_source_signal ? "yes" : "no");
+		syslog_server(priority, "\tis destination? %s", sig_info->is_destination_signal ? "yes" : "no");
+		syslog_server(priority, "\tindex in path   %d", sig_info->index_in_route_path);
```

Comment:
I see why, but line breaks in one syslog message are problematic because of formatting (I think?) and one big long line I would consider to be unreadable. Not sure what I would prefer here.

Reply:
How long would the message be?  
`Drive route signal info: id - signal1, not source, not destination, path index - 2`

```c
syslog_server(priority, "Drive route signal info: id - %s, %s source, %s destination, path index - %d", 
  sig_info->id != NULL ? sig_info->id : "NULL",
  sig_info->is_source_signal ? "is" : "not",
  sig_info->is_destination_signal ? "is" : "not",
  sig_info->index_in_route_path)
```

Reply:
oh, right - I thought about it if applied to all signal-info prints of the array inline. If its one then that is not that long yeah. will be adjusted